### PR TITLE
[MM-37314] Fix the categorization bug

### PR DESCRIPTION
**Pull request by iomodo**
#### Summary
An ugly fix for the categorization bug.
Comment in the code is pretty self-explanatory.
When user starts a run, channel is created, and user added to the channel. At that moment server sends the `user_added` socket event to the webapp. Webapp gets the event and tries to categorize the channel for the user by adding the channel to the default `CHANNELS` category. 
Concurrently we are trying to auto-categorize the channel into our category. Depending on what happens first channel might get stuck in `CHANNELS` category.

Unfortunately we can not reliably detect when webapp finished updating the state, that's why we are waiting for 5 seconds(A very magic number) to let webapp do its job.

#### Ticket Link
  Fixes: https://mattermost.atlassian.net/browse/MM-37314

#### Checklist

- ~~[ ] Telemetry updated~~
- ~~[ ] Gated by experimental feature flag~~
- ~~[ ] Unit tests updated~~

### CODE REVIEW

**Review by agarciamontoro — APPROVED** [review]
I'm ok merging this if we want it for 1.20, but we should make sure to apply a proper fix as soon as we can. Can we maybe create a follow-up ticket for this and continue the investigation?

File: server/app/playbook_run_service.go [hunk 1]
```diff
@@ -1703,6 +1703,15 @@ func (s *PlaybookRunServiceImpl) UserHasJoinedChannel(userID, channelID, actorID
...
 // it does not already exist and adds the channel within the sidebar category
 func (s *PlaybookRunServiceImpl) createOrUpdatePlaybookRunSidebarCategory(userID, channelID, teamID, categoryName string) error {
+	// Wait for 5 seconds(a magic number) for the webapp to get the `user_added` event,
+	// finish channel categorization and update it's state in redux.
+	// Currently there is no way to detect when webapp finishes the job.
+	// After that we can update the categories safely.
+	// Technically if user starts multiple runs simultaneously we will still get the race condition
+	// on category update. Since that's not realistic at the moment we are not adding the
+	// distributed lock here.
+	time.Sleep(5 * time.Second)
```
Comment (lieut-data) [inline server/app/playbook_run_service.go:1713]: Let's not actually block the completion of the hook here, though, and run instead spawn a goroutine that blocks for the required workaround time.

**Review by lieut-data — CHANGES_REQUESTED** [review]
Nice investigation :D

One comment below.

**Review by lieut-data — APPROVED** [review]

**Comment by lieut-data** [comment]
@iomodo, I /think/ the failures here are due to my requested changes re: making the categorization happen async: I guess the unit tests need to be updated?

**Review by lieut-data — APPROVED** [review]

**Pull request merged**